### PR TITLE
Add support for device authorization flow (RFC8628)

### PR DESCRIPTION
**Pull request by kellyma2**
This change adds support for device authorization flow with a new DeviceClient implementation that is derived somewhat from MobileApplicationClient and BackendApplicationClient. 

DeviceClient provides a method to:

(1) generate the request URI for device code and authorization URLs

(2) query the token endpoint with a provided device code to retrieve the OAuth token

Additionally, a unit test is provided in order to validate the DeviceClient operations

### CODE REVIEW

File: oauthlib/oauth2/rfc8628/clients/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
```
Comment (auvipy) [inline oauthlib/oauth2/rfc8628/clients/__init__.py:1]: this 
# -*- coding: utf-8 -*- line is not needed anymore on python3+

File: tests/oauth2/rfc8628/clients/test_device.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
```
Comment (auvipy) [inline tests/oauth2/rfc8628/clients/test_device.py:1]: remove utf-8 line please

**Review by auvipy — CHANGES_REQUESTED** [review]

**Comment by auvipy** [comment]
overall looks good

File: oauthlib/oauth2/rfc8628/clients/__init__.py (hunk 1 above)
Comment (kellyma2) [reply oauthlib/oauth2/rfc8628/clients/__init__.py]: Done

File: tests/oauth2/rfc8628/clients/test_device.py (hunk 2 above)
Comment (kellyma2) [reply tests/oauth2/rfc8628/clients/test_device.py]: done

**Review by thedrow — APPROVED** [review]

**Comment by kellyma2** [comment]
@auvipy are you able to go through this again? also @jonathanhuot?

**Review by auvipy — APPROVED** [review]

**Pull request merged**